projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -395,12 +395,24 @@ public function render_block_item( $related_post, $block_attributes ) {
 			$item_markup .= $date_tag;
 		}
 
-		if ( ( $block_attributes['show_context'] ) && ! empty( $related_post['context'] ) ) {
-			$context_tag = sprintf(
-				'<li class="jp-related-posts-i2__post-context">%1$s</li>',
-				esc_html( $related_post['context'] )
-			);
-
+		if ( ( $block_attributes['show_context'] ) && ! empty( $related_post['block_context'] ) ) {
+			// Note: The original 'context' value is not used when rendering the block.
+			// It is still generated and available for the legacy rendering code path though.
+			// See './related-posts.js' for that usage.
+			$context_tag   = '';
+			$block_context = $related_post['block_context'];
+			if ( ! empty( $block_context['link'] ) ) {
+				$context_tag = sprintf(
+					'<li class="jp-related-posts-i2__post-context"><a href="%1$s">%2$s</a></li>',
+					$block_context['link'],
+					$block_context['text']
```

Comment:
Do we need to do some escaping here? It may be overkill, but I'm wondering if it may be useful in some cases.

Reply:
My understanding is that it's not needed since both values are pulled from core functions but I don't think it hurts to add it. Incoming!

> most WordPress functions properly prepare the data for output, and additional escaping is not needed.

https://developer.wordpress.org/apis/security/escaping/

Reply:
Also, (a) the rest of the function is escaping data without concern for the origin and (b) easier for readers of the function if they don't have to double-check how the data is generated.